### ISSUE TITLE
Dequantize all bnb.nn.Linear8bitLt in a model

### ISSUE THREAD

**Issue by vince62s** [TimDettmers/bitsandbytes#219]
Hello,

Thanks for this great package.
Once I have finetuned a model for which I have replaced all nnLinear by bnb.nn.Linear8bitLt, how can I revert back to dequantized layers in an easy way to save the model back to fp16 ?

Thanks.

**Comment by vince62s** [comment]
I guess this is it:
(int8_model[0].weight.CB * int8_model[0].weight.SCB) / 127

**Comment by entrpn** [comment]
I have been working on this as well and the problem is that CB and SCB will be None after the first forward pass. I'm pretty new at this and I been looking at the code and it seems this might do the trick but I'm still not sure:

```python
CxB = layer.state.CxB
SCB = layer.state.SCB
fp16_weights = ((CxB * SCB) / 127)
```

**Comment by vince62s** [comment]
Second thought on this. I guess you want to finetune (ie: train mode). I am not sure the best practice to finetune is to do it all int8. I think it requires the `has_fp16_weights` but I am not 100% sure.
The issue with this flag being True, it will take way more memory during finetuning.

**Comment by entrpn** [comment]
@vince62s here is an [example](https://github.com/entrpn/serving-model-cards/tree/main/instruct-tune-gptj) of how to finetune GPTJ on a T4 GPU. Hope it helps.
